### PR TITLE
[BANKCON-11625] FC - Theme textfields

### PR DESCRIPTION
**Pull request by mats-stripe**
## Summary

This themes the financial connections `RoundedTextField` highlighted state, based on the manifest's theme.

## Motivation

Alignment with the [Instant Debits figma](https://www.figma.com/design/wXQ8NJApqSJiLmxxANDRTs/%E2%9C%A8-Connections-3.0?node-id=791-99467&t=HT7VHACA6DyEV2Ly-4)

## Testing

| Financial Connections | Instant Debits |
|--------|--------|
| Simulator Screenshot - iPhone 15 - 2024-07-22 at 16 22 47 | Simulator Screenshot - iPhone 15 - 2024-07-22 at 16 24 03 | 

## Changelog

N/a

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift [hunk 1]
```diff
@@ -100,7 +101,7 @@ final class RoundedTextField: UIView {
         var theme: ElementsUITheme = .default
         theme.colors = {
             var colors = ElementsUITheme.Color()
-            colors.primary = .brand500
+            colors.primary = self.theme.textFieldFocusedColor
```
Comment (kgaidis-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift:104]: @mats-stripe 

Is that expected that this changed from `.brand500` to `.brand600`?

**Review by kgaidis-stripe — DISMISSED** [review]

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift [hunk 2]
```diff
@@ -675,37 +685,70 @@ struct RoundedTextField_Previews: PreviewProvider {
...
+            RoundedTextFieldUIViewRepresentable(
+                placeholder: "Link Light theme",
+                footerText: nil,
+                errorText: nil,
+                isFocused: true,
+                theme: .linkLight
+            )
+            .frame(height: 56)
+            .padding()
+            .previewDisplayName("Focused - Link Light theme")
```
Comment (mats-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift:751]: In order to showcase the different focused state themes, we had to use separate SwiftUI preview devices:

https://github.com/user-attachments/assets/83c6ad99-69d3-4b56-aa21-c0498ab7427f

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift (hunk 1 above)
Comment (mats-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift:104]: Whoops - good catch!

**Review by kgaidis-stripe — DISMISSED** [review]

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift (hunk 1 above)
Comment (mats-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/RoundedTextField.swift]: Fixed in https://github.com/stripe/stripe-ios/pull/3813/commits/cd5a6d409abb53ad71571d277e41bf16de09e4b5!

**Review by kgaidis-stripe — APPROVED** [review]

**Pull request merged**